plugins/Diagnostics/Diagnostic/AbstractPrivateDirectories.php

```diff
@@ -88,7 +89,22 @@ private function getUrlList(array $testUrls)
 
     protected function isAccessible(DiagnosticResult $result, $testUrl, $publicIfResponseEquals, $publicIfResponseContains)
     {
+        static $addedDisableHttpDiagnosticsWarning = false;
```

Comment:
Maybe put this logic in the `execute` method right after the check whether internet is disabled? Then you won't need to use `static` (which can cause issues in tests etc) and it will be executed only one for each private directory check. You can then see `This check was skipped as this check is disabled in the config. To enable this check set [...]... in the "config/config.ini.php" file.`

Reply:
Not 100% sure I implemented what you mean here. Please check below.

Reply:
@geekdenz I mentioned to put it into the `execute` method right after the check whether internet is disabled. That's where it needs to go so you don't need to check if it has been triggered before etc and this way others will know what's happening etc. Could you check if that makes sense?

Reply:
Ah! Sorry. I think I understood now what you mean. Will do.

Reply:
Still running Travis tests so not taken out of draft, but if you have time and patience, you could check if i implemented what you meant, @tsteur . Locally it works.

Reply:
👍  haven't tested it but that looks better. You won't need `$this->addedDisableHttpDiagnosticsWarning` anymore now and fyi you're mentioning in the ini description that it runs the checks on "install" but in the beginning of the execute method we're making sure that we don't run these tests on install as they wouldn't make quite sense yet as some of these files wouldn't exist just yet etc. Meaning you will want to change eg `Enable HTTP checks on install or Diagnostics` to `Enable HTTP checks in the diagnostic system check`